ci-automation/tapfile_helper_lib.sh

```diff
@@ -77,69 +77,92 @@ function __db_init() {
        __db_init
     fi
 
     # Wrap all SQL commands in a transaction to speed up INSERTs
+    # We will commit intermediately if there's an error message to insert so the
+    # error message file can be reused.
     local SQL="BEGIN TRANSACTION;"
+    local has_error_message="false"
 
-    # Example TAP input:
-    # ok - coreos.auth.verify
-    # ok - coreos.locksmith.tls
-    # not ok - cl.filesystem
-    #   ---
-    #   Error: "--- FAIL: cl.filesystem/deadlinks (1.86s)\n            files.go:90: Dead symbolic links found: [/var/lib/flatcar-oem-gce/usr/lib64/python3.9/site-packages/certifi-3021.3.16-py3.9.egg-info]"
-    #   ...
-    # ok - cl.cloudinit.script
-    # ok - kubeadm.v1.22.0.flannel.base
-    while read -r line; do
-        if [[ "${line}" == "1.."* ]] ; then continue; fi
-        if [ "${line}" = "---" ] ; then  # note: read removes leading whitespaces
-            in_error_message=true
-            continue
-        fi
 
-        if $in_error_message ; then
-            if [ "${line}" = "..." ] ; then
-                in_error_message=false
-            else
-                error_message="$(echo -e "$line" \
-                                    | sed -e 's/^Error: "--- FAIL: /"/' -e 's/^[[:space:]]*//' \
-                                    | sed -e "s/[>\"']/_/g" -e 's/[[:space:]]/ /g')"
+    # run the parse loop in a subshell and clean up temporary error message file on exit
+    (
+        local error_message_file="$(mktemp)"
+        trap "rm -f '${error_message_file}'" EXIT
+        # Example TAP input:
+        # ok - coreos.auth.verify
+        # ok - coreos.locksmith.tls
+        # not ok - cl.filesystem
+        #   ---
+        #   Error: "--- FAIL: cl.filesystem/deadlinks (1.86s)\n            files.go:90: Dead symbolic links found: [/var/lib/flatcar-oem-gce/usr/lib64/python3.9/site-packages/certifi-3021.3.16-py3.9.egg-info]"
+        #   ...
+        # ok - cl.cloudinit.script
+        # ok - kubeadm.v1.22.0.flannel.base
+        while read -r line; do
+            if [[ "${line}" == "1.."* ]] ; then continue; fi
+            if [ "${line}" = "---" ] ; then  # note: read removes leading whitespaces
+                in_error_message=true
                 continue
             fi
-        else
-            test_name="$(echo "${line}" | sed 's/^[^-]* - //')"
-            local result_string
-            result_string="$(echo "${line}" | sed 's/ - .*//')"
-            result=0
-            if [ "${result_string}" = "ok" ] ; then
-                result=1
+
+            if $in_error_message ; then
+                if [ "${line}" = "..." ] ; then
+                    in_error_message=false
+                    has_error_message="true"
+                else
+                    echo -e "$line" \
+                        | sed -e 's/^Error: "--- FAIL: /"/' -e 's/^[[:space:]]*//' \
+                              -e "s/[>\"']/_/g" -e 's/[[:space:]]/ /g' \
+                              -e 's/.\{200\}/&\n/g' \
```

Comment:
Maybe replacing literal `\n` with newlines could also help with overlong lines - those seem to exist even in the example tap output above.

```suggestion
                              -e "s/[>\"']/_/g" -e 's/[[:space:]]/ /g' \
                              -e 's/\\n/\n/g' \
                              -e 's/.\{200\}/&\n/g' \
```